sopel/modules/reddit.py

```diff
@@ -189,14 +196,57 @@ def comment_info(bot, trigger, match):
 
 
 # If you change this, you'll have to change some other things...
-@commands('redditor')
-@example('.redditor poem_for_your_sprog')
-def redditor_info(bot, trigger, match=None):
+def subreddit_info(bot, trigger, match, is_command=False):
+    """Shows information about the given subreddit"""
+    r = bot.memory['reddit_praw']
+    try:
+        r.subreddits.search_by_name(match, exact=True)
+    except prawcore.exceptions.NotFound:
+        if is_command:
+            bot.say('No such subreddit.')
+        # Fail silently if it wasn't an explicit command.
+        return NOLIMIT
+
+    try:
+        r.subreddit(match).subreddit_type
+    except prawcore.exceptions.Forbidden:
+        bot.say("r/" + match + " appears to be a private subreddit!")
+        return NOLIMIT
+    except prawcore.exceptions.NotFound:
+        bot.say("r/" + match + " appears to be a banned subreddit!")
+        return NOLIMIT
+
+    s = r.subreddit(match)
+    link = "https://www.reddit.com/r/" + s.display_name
+
+    created = get_time_created(bot, trigger, s.created_utc)
+
+    message = ('[REDDIT] {link}{nsfw} | {subscribers} subscribers | '
+               'Created at {created} | {public_description}')
+
+    nsfw = ''
+    if s.over18:
+        nsfw += ' ' + bold(color('[NSFW]', colors.RED))
+
+        sfw = bot.db.get_channel_value(trigger.sender, 'sfw')
+        if sfw:
+            link = '(link hidden)'
+            bot.kick(
+                trigger.nick, trigger.sender,
+                'Linking to NSFW content in a SFW channel.'
+            )
+
+    message = message.format(
+        link=link, nsfw=nsfw, subscribers='{:,}'.format(s.subscribers),
+        created=created, public_description=s.public_description)
+    bot.say(message)
+
+
+def redditor_info(bot, trigger, match, is_command=False):
     """Shows information about the given Redditor"""
-    commanded = re.match(bot.config.core.prefix + 'redditor', trigger)
-    match = match or trigger
+
```

Comment:
No blank line at the beginning of a function. (Someday, `checkstyle.sh` should evolve to enforce things like this that aren't technically PEP8 violations.

Reply:
Unresolved this because it was untouched.

Python docstring conventions dictate there should be no blank lines either before or after. Both `redditor_info` and `subreddit_info` have a blank line after their respective docstrings, and shouldn't.